HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,123 @@
+import pyspark
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+
+def link_string_to_KV(s):
+    src, dests = s.split(': ')
+    dests = [int(to) for to in dests.split(' ')]
+    return (int(src), dests)
+
+
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+
+def ssbfs(neighbors,page_names,inspect=[]):
+	neighbors=neighbors.cache()
+
+    originPage=inspect[0] # Passing in the node ID's for the origin
+    endPage=inspect[1] # Passing in the node ID for the goal end page
+
+ 	Found=False
+
+	newNeighbors=neighbors.filter(lambda (k,v): k == originPage).flatMapValues(lambda x: x)
+	newNeighborsList=newNeighbors.map(lambda (k,v): v).collect()
+	Found=endPage in rightNeighborsList
+
+	oldNeighborsList=newNeighborsList
+
+	allNeighbors_touched=[]
+	allNeighbors_touched.append(newNeighborsList)
+	ii=1
+    while Found is False:
+    	ii=ii+1
+    	newNeighbors=neighbors.filter(lambda (k,v): k in oldNeighborsList).flatMapValues(lambda x: x)
+		newNeighborsList=newNeighbors.map(lambda (k,v): v).collect()
+		allNeighbors_touched.append(newNeighborsList)
+		Found= endPage in newNeighborsList
```

Comment:
This search is done in python.
You could have done this in spark, by using joins and subtractByKey.